src/output_writers.jl

```diff
@@ -270,126 +270,215 @@ end
+zdim(::Type{Cell}) = "zC"
+
+# This function converts and integer to a range, and nothing to a Colon
+get_slice(n::Integer) = n:n
+get_slice(n::UnitRange) = n
+get_slice(n::Nothing) = Colon()
+
+"""
+    write_grid(model; filename="grid.nc", mode="c", 
+               compression=0, attrib=Dict(), slice_kw...)
+
+Writes a grid.nc file that contains all the dimensions of the domain.
+
+Keyword arguments
+=================
+
+    - `filename::String`  : File name to be saved under
+    - `mode::String`      : Netcdf file is opened in either clobber ("c") or append ("a") mode (Default: "c" )
+    - `compression::Int`  : Defines the compression level of data (0-9, default 0)
+    - `attrib::Dict`      : Attributes (default: Dict())
+"""
+function write_grid(model; filename="./grid.nc", mode="c",
+                    compression=0, attrib=Dict(), slice_kw...)
+    dimensions = Dict(
         "xC" => collect(model.grid.xC),
         "yC" => collect(model.grid.yC),
         "zC" => collect(model.grid.zC),
-        "xF" => collect(model.grid.xF),
-        "yF" => collect(model.grid.yF),
-        "zF" => collect(model.grid.zF),
-        "u" => Array(parentdata(model.velocities.u)),
-        "v" => Array(parentdata(model.velocities.v)),
-        "w" => Array(parentdata(model.velocities.w)),
-        "T" => Array(parentdata(model.tracers.T)),
-        "S" => Array(parentdata(model.tracers.S))
+        "xF" => collect(model.grid.xF)[1:end-1],
+        "yF" => collect(model.grid.yF)[1:end-1],
+        "zF" => collect(model.grid.zF)[1:end-1]
+    )
+    dim_attrib = Dict(
+        "xC" => ["longname" => "Locations of the cell centers in the x-direction.", "units" => "m"],
+        "yC" => ["longname" => "Locations of the cell centers in the y-direction.", "units" => "m"],
+        "zC" => ["longname" => "Locations of the cell centers in the z-direction.", "units" => "m"],
+        "xF" => ["longname" => "Locations of the cell faces in the x-direction.",   "units" => "m"],
+        "yF" => ["longname" => "Locations of the cell faces in the y-direction.",   "units" => "m"],
+        "zF" => ["longname" => "Locations of the cell faces in the z-direction.",   "units" => "m"]
     )
 
-    if fw.async
-        # Execute asynchronously on worker 2.
-        i = model.clock.iteration
-        @async remotecall(write_output_netcdf, 2, fw, fields, i)
-    else
-        write_output_netcdf(fw, fields, model.clock.iteration)
+    # Applies slices to the dimensions d
+    for (d, slice) in slice_kw
+        if String(d) in keys(dimensions)
+            dimensions[String(d)] = dimensions[String(d)][get_slice(slice)]
+        end
     end
 
+    # Writes the sliced dimensions to the specified netcdf file
+    Dataset(filename, mode, attrib=attrib) do ds
+        for (dimname, dimarray) in dimensions
+            defVar(ds, dimname, dimarray, (dimname,),
+                   compression=compression, attrib=dim_attrib[dimname])
+        end
+    end
     return
```

Comment:
I think we agreed to use `return nothing`? @ali-ramadhan ? Not a big deal obviously.

Reply:
Fixed.